subsys/net/l2/openthread/openthread.c

```diff
@@ -96,6 +96,18 @@ static struct net_linkaddr *ll_addr;
 
 static struct net_mgmt_event_callback ip6_addr_cb;
 
+#include <power/reboot.h>
+#include <stdio.h>
+int devicedata_get_pskd(const char **);
```

Comment:
@rlubos: will you make that openthread API or should we make a suggestion how it could look like?

we are reworking the rest of the code so it will take a bit till i will push it again. Thanks for your help!

Reply:
I already suggested the rename: `openthread_get_psdd`. By making it available in the API I just meant that we should put the declaration in `include/net/openthread.h`, a public header that is available for applications. Users need to know what function they should implement.

Reply:
I'm actually confused... why does it need to be declared in openthread.h in any case? I see it if it was openthread_SET_pskd(), which would be called by app... but currently it works the other way around

Reply:
Yes, it works the other way, but still, it needs to be documented how the function should look like. How is the user supposed to know what function should he implemented and how should it look like, if we don't provide the expected function declaration?